app/src/main/java/com/mobileproto/hireddit/hireddit/GetWordsAsync.java

```diff
@@ -0,0 +1,120 @@
+package com.mobileproto.hireddit.hireddit;
+
+import android.content.Context;
+import android.content.res.AssetManager;
+import android.os.AsyncTask;
+import android.util.Log;
+import android.widget.TextView;
+
+import com.google.common.io.CharStreams;
+
+import java.io.IOException;
+import java.io.InputStream;
+import java.io.InputStreamReader;
+import java.util.ArrayList;
+
+import io.indico.Indico;
+import io.indico.network.IndicoCallback;
+import io.indico.results.IndicoResult;
+import io.indico.utils.IndicoException;
+
+;
+
+/**
+ * Created by lwilcox on 11/5/2015.
+ */
+public class GetWordsAsync extends AsyncTask<Void, Void, ArrayList<String>>
+{
+    private String spokenString;
+    private String importantWords;
+    private Context context;
+    public ArrayList<String> wordList = new ArrayList<>();
+    public ArrayList<String> allComments;
+    public String postComment;
+    public TextView commentText;
+    public Indico indico;
+
+
+    public GetWordsAsync(String spokenString, String importantWords, Context context, TextView commentText)
+    {
+
+        this.spokenString = spokenString;
+        this.importantWords = importantWords;
+        this.context = context;
+        this.commentText = commentText;
+        String indicoApiKey = getApi(context);
+        this.indico = Indico.init(context, indicoApiKey, null);
+
+    }
+
+    public static String getApi(Context context)
+    {
+        try {
+            AssetManager assetManager = context.getAssets();
+            InputStream apikey = assetManager.open("indicoapitxt.txt");
+            String apiKeyString = CharStreams.toString(new InputStreamReader(apikey, "UTF-8"));
+            Log.d("apikey", apiKeyString);
+            return apiKeyString;
+        } catch (IOException e) {
+            e.printStackTrace();
+            return null;
+        }
+    }
+
+    @Override
+    protected ArrayList<String> doInBackground(Void... params)
+    {
+        try {
+            //final ArrayList<String> wordList = new ArrayList<String>();
+            indico.keywords.predict(spokenString, new IndicoCallback<IndicoResult>()
+            {
+                @Override
+                public void handle(IndicoResult result) throws IndicoException
+                {
+                    Log.i("Indico Keywords", "keywords: " + result.getKeywords());
+                    if (result.getKeywords() != null) {
+                        wordList.add(result.getKeywords().keySet().toString());
+                    }
+                }
+            });
+            Log.d("wordlist", wordList.toString());
+            while (wordList.isEmpty()) {
+                Log.d("wating", "true");
+                try {
+                    Thread.sleep(15);
```

Comment:
probably want to make the time shorter/show a toast if its waiting for `X` seconds saying something then after the full time is true show toast saying question didn't work. also, in the case that indico fails us, do we just search the full string on reddit?